packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -151,11 +152,12 @@ export class BaseSnapExecutor {
           result = null;
         }
 
-        assert(
-          isValidJson(result),
-          new TypeError('Received non-JSON-serializable value.'),
-        );
-        return result;
+        // /!\ Always return only sanitized JSON to prevent security flaws. /!\
+        try {
+          return getSafeJson(result);
+        } catch {
+          throw new TypeError('Received non-JSON-serializable value.');
```

Comment:
Can't we just use the error from `getSafeJson`? It should be a readable Superstruct error.

Reply:
How would we do that? Doesn't `create()` throw?

Reply:
Updated. It will use the error coming from the `getSafeJson`. One test is affected by this change, so I updated that as well.

Reply:
@Mrtenz I find the previous error more readable tbh. What do you think?

Reply:
Let's combine the two errors:
```
throw new TypeError(`Received non-JSON-serializable value: ${error.message}`);
```

Reply:
Might need a `instanceof`:

```ts
if (error instanceof StructError) {
  throw new TypeError(`Received non-JSON-serializable value: ${error.message}`);
}

throw error;
```

Reply:
If JSON.parse / JSON.stringify fails it won't be a StructError though

Reply:
Added errors combined.